FEA Add metadata routing to SelectFromModel (#27490)
scikit-learn/scikit-learn: doc/whats_new/v1.4.rst

  ``**score_params`` which are passed to the underlying scorer.
  :pr:`26525` by :user:`Omar Salman <OmarManzoor>`.

> - |Feature| :class:`feature_selection.SelectFromModel` now supports metadata
>   routing according to :ref:`metadata routing user guide <metadata_routing>`.

Reviewer: ```suggestion
- |Feature| :class:`feature_selection.SelectFromModel` now supports metadata
  routing in `fit` and `partial_fit`.
```
